packages/inter-protocol/test/reserve/reserve.test.js

```diff
@@ -77,9 +77,159 @@ test('reserve add collateral', async t => {
   );
 });
 
-test('check allocations', async t => {
+test('reserve withdraw collateral', async t => {
+  const atomKit = makeIssuerKit('atom');
   /** @param {NatValue} value */
+  const atom = value => AmountMath.make(atomKit.brand, value);
+
+  const electorateTerms = { committeeName: 'EnBancPanel', committeeSize: 3 };
+  const timer = buildZoeManualTimer(t.log);
+
+  const { zoe, reserve } = await setupReserveServices(
+    t,
+    electorateTerms,
+    timer,
+  );
+
+  await E(reserve.reserveCreatorFacet).addIssuer(atomKit.issuer, 'ATOM');
+  const invitation = await E(
+    reserve.reservePublicFacet,
+  ).makeAddCollateralInvitation();
+
+  const proposal = { give: { Collateral: atom(100_000n) } };
+  const atomPayment = atomKit.mint.mintPayment(atom(100000n));
+  const payments = { Collateral: atomPayment };
+  const collateralSeat = E(zoe).offer(invitation, proposal, payments);
+
+  t.is(
+    await E(collateralSeat).getOfferResult(),
+    'added Collateral to the Reserve',
+    `added atom to the collateral Reserve`,
+  );
+
+  const makeWithdrawInvitationMakers = async () => {
+    const continuingInvitation = await E(
+      reserve.reserveCreatorFacet,
+    ).makeRepeatableWithdrawalInvitation();
+    const { invitationMakers } = await E(
+      E(zoe).offer(continuingInvitation),
+    ).getOfferResult();
+    return invitationMakers;
+  };
+
+  t.log('making cached repeated withdrawal makers');
+  const cachedMakers = await makeWithdrawInvitationMakers();
+
+  const freshSingleWithdrawal = () =>
+    E(reserve.reserveCreatorFacet).makeSingleWithdrawalInvitation();
+  const freshRepeatedWithdrawal = () =>
+    makeWithdrawInvitationMakers().then(im => E(im).Withdraw());
+  const cachedRepeatedWithdrawal = () => E(cachedMakers).Withdraw();
+
+  const revokedRegExp =
+    /Withdraw Collateral Invitation has been cancelled|WithdrawalFacet.* revoked/;
+
+  const withdrawMakers = {
+    cachedRepeatedWithdrawal,
+    freshRepeatedWithdrawal,
+    freshSingleWithdrawal,
+  };
```

Comment:
Did you consider using an ava macro to make these independent tests?
mostly just curious; not worth the bother at this point